sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -721,6 +720,11 @@ abstract class RapidsShuffleThreadedReaderBase[K, C](
         shuffleReadTimeNs.foreach(_ += shuffleReadTime)
         dataReadSize.foreach(_ += uncompressedSize)
 
+        // Reset fetch, readBlocked, and wait times to prevent extraneous accruals
+        waitTime = 0L
+        fetchTime = 0L
```

Comment:
the problem was, as you described for `waitTime` that the variable is a class member instead of scoped. It is set the first time there, and it is not updated when there isn't a wait to 0. So if we make this scoped like the others, I think that takes care of the issue as well.

Reply:
Changed in latest commit to have accumulators update when sub-metric accruals happen